o-fish-ios/ViewModel/Report/LocationViewModel.swift

```diff
@@ -35,25 +36,14 @@ class LocationViewModel: ObservableObject, Identifiable {
         self.location = location
     }
 
-    convenience init(_ location: Location?) {
+    convenience init(_ location: List<Double>?) {
         self.init()
         if let location = location {
             locationModel = location
-            latitude = location.latitude
-            longitude = location.longitude
+            latitude = location[1]
```

Comment:
FYI this is unsafe operation and will crash if array have no element for index 0 and 1

consider using
```
location.last ?? 0
location.first ?? 0
```